client/src/main/java/io/pravega/client/segment/impl/SegmentOutputStreamImpl.java

```diff
@@ -305,7 +308,7 @@ public void wrongHost(WrongHost wrongHost) {
         @Override
         public void segmentIsSealed(SegmentIsSealed segmentIsSealed) {
             log.info("Received SegmentSealed {} on writer {}", segmentIsSealed, writerId);
-            if (state.sealEncountered.compareAndSet(false, true)) {
+            if (state.encounteredSeal()) {
```

Comment:
On receiving a Segment Sealed `EventStreamWriterImpl#handleLogSealed` is invoked. The `SegmentoutputStream` executes it on  `connectionFactory.getInternalExecutor()` with `Retry.indefinitelyWithExpBackoff` .

`handleLogSealed` internally executes it on `retransmitPool` with `Retry.indefinitelyWithExpBackoff`. This causes the thread of `connectionFactory.getInternalExecutor()` which is a limited resource to be blocked until `handleLogSealed` is completed.

Reply:
>handleLogSealed internally executes it on retransmitPool with Retry.indefinitelyWithExpBackoff

yes. it is on retransmitPool, and the thread in connectionFactory.getInternalExecutor() is never blocked.

Reply:
ok, got it..